### ISSUE TITLE
enhance router_check_tool to allow going through listeners as well

### ISSUE THREAD

**Issue by howardjohn** [envoyproxy/envoy#12393]
Currently, `router_check_tool` takes in a route as input. This makes the tool somewhat whitebox - the user first needs to figure out which route they care about, then they can write a test for it. Ideally, its completely black box - the user gives the full Envoy config (really just Listeners and Routes) and a set of queries, and checks the result.

This is useful as well to test complex listeners with many different filter chain matches.

Alternatives considered:
* Implement the logic for determining a route from the request ourselves. This would involve re-implementing the filter chain/listener matching logic which is likely to be incorrect and fall out of sync
* Use the tool as is. This doesn't mean the requirement of black box testing, relies test writers to know Envoy implementation details
* Send real traffic against a real envoy instead. This is slower and more complex

**Comment by mattklein123** [comment]
cc @LisaLudique who might have some thoughts here.

**Comment by howardjohn** [comment]
Especially useful in light of the complexity around FCM matching logic: https://github.com/envoyproxy/envoy/issues/12572

**Comment by howardjohn** [comment]
Not stale, I think this would be really useful.

**Comment by mattklein123** [comment]
@howardjohn I realize now this is very similar to this issue I opened? https://github.com/envoyproxy/envoy/issues/13065

**Comment by howardjohn** [comment]
Very related. Basically the goal is instead of having individual route/extauth/rbac tools we just have a single one that takes in a full envoy config. This way we can write completely black box tests

**Comment by mattklein123** [comment]
Yes makes sense. We should look at both issues together and figure out the right path forward. Maybe now that we have the filter manager split out from @snowp we could figure out a way of driving a synthetic requests/response through the filter manager and inspecting the state at various points.
